### ISSUE TITLE
getting better at open source: make mathsteps easier to contribute to and maintain

### ISSUE THREAD

**Issue by evykassirer** [socraticorg/mathsteps#134]
I'd like to continue to make `mathsteps` better in the open source community: more people finding us and wanting to help, easier for people to get started, fun for people to keep contributing, easier for maintainers to maintain, etc.

A lot of these tasks can be done by anyone, so if you'd like to help out lemme know!

I found https://opensource.guide/ which looks amazing, and am taking notes on stuff I'd like to do:

- [ ] Update README
   - make it clearer what mathsteps does and why it's helpful
   - acknowledgements to author/contributors/mathjs
   - link to the further documentation in the wiki
   - more resources: https://pages.18f.gov/open-source-guide/making-readmes-readable/ and https://gist.github.com/PurpleBooth/109311bb0361f32d87a2

- [ ] document npm [releases](https://github.com/socraticorg/mathsteps/releases) on github by using [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) (e.g. https://github.com/PurpleBooth/git-lint-validators/releases)

- [ ] update CONTRIBUTING
   - should we also put installation instructions here (it's already in the README) ?
   - link to the roadmap from the wiki
   - add some more stuff from https://github.com/nayafia/contributing-template/blob/master/CONTRIBUTING-template.md and http://mozillascience.github.io/working-open-workshop/contributing/

- [ ] [issue and pull request templates](https://github.com/blog/2111-issue-and-pull-request-templates) if we find certain a certain structure that makes sense and would help (I can't think of one right now)

Read more:

- [x] https://opensource.guide/starting-a-project/
- [ ] https://opensource.guide/finding-users/
- [ ] https://opensource.guide/building-community/
- [ ] https://opensource.guide/best-practices/
- [ ] https://opensource.guide/leadership-and-governance/
- [ ] maybe https://opensource.guide/getting-paid/
- [ ] https://opensource.guide/metrics/
- [ ] look into ways to support contributors and help them become long term contributors

**Comment by nitin42** [comment]
I will look into this !

**Comment by evykassirer** [comment]
oh, and feel free to comment here anything else to add to the list!

**Comment by nitin42** [comment]
I think I can make a github page or something for enhancing the wiki ! What do you think?

**Comment by evykassirer** [comment]
what do you mean by a "github page"? and I'd love to enhance the wiki :)

**Comment by nitin42** [comment]
I mean setting up a gitbook like thing (see [this](https://developmentarc.gitbooks.io/react-indepth/)) or setting up a static page (using Jekyll and Github).

**Comment by nitin42** [comment]
If you want to keep wiki compact then I'd say we should go for [GitHub Pages](https://pages.github.com/) or for a more detail explanation GITBOOK is awesome......!!!

**Comment by evykassirer** [comment]
whoaaa gitbook looks great! let's move this conversation to #66

**Comment by nitin42** [comment]
@evykassirer I've made some changes to the PR (`mathsteps-demo`) and I was thinking of adding the docs and contributing guide to website. Let me know what you think!

**Comment by evykassirer** [comment]
ah sorry Nitin! Let me look at your current PR - I've been a bit busy but I'll look either tonight or in the next day or two.

Did you want to add the current docs or write some new ones? If we're doing something new, let's talk about it first to make sure we're on the same page :)

**Comment by nitin42** [comment]
If we're rewriting the core than it doesn't makes sense to render the old docs. I want to jump in where new changes are happening. Let me know where I can start! Also if there is some other stuff or some other issues to work on, let me know! I'll be happy to contribute more 😀!

**Comment by evykassirer** [comment]
sounds good!

I'm gonna explore what we need to do for the refactoring stuff within the week (hopefully!) and we can get started from there - I'm still not totally sure what the work will involve and what the issues that need tackling will be, and it'd be nice to have more concrete tasks before we start working on it - and it'd be nice to get a few people taking on tasks if possible so I think you can help with it!

In the meantime, you could check out the semantic math repo and start familiarizing yourself with it - especially math-parser and math-rules. We might need to add some stuff there too, and there's a bunch of open issues there as well.

**Comment by nitin42** [comment]
I was thinking about replacing the current test suite with Jest!

**Comment by evykassirer** [comment]
Sorry for the delay - I'm happy to look into that once I'm done with the refactor (hopefully will finish in the next two weeks, I'm planning to sink a bunch of time into it!)

Thanks for suggesting it - I'll open an issue for now :)

**Comment by nitin42** [comment]
Welcome back! 😜
